### PR TITLE
Fix app section

### LINKED ISSUE #27344: Signed app tarball still appears experimental
Steps:

1. Download owncloud daily master (tested with March 9th 2017)
1. Download the contacts app tarball from https://apps.owncloud.com/content/show.php/Contacts?content=168708
1. Change appinfo/info.xml and set the max version to "10.0" instead of "9.2"
1. Setup OC
1. Go to Settings/Apps and click "Disabled apps"
1. Scroll to the contacts app

### Expected
Contacts app appears approved

### Actual
Contacts app appears as experimental.

### Versions
ownCloud daily master and git.

I did the same test with a 9.1.4 tarball and the contacts app appears "Approved" there.

@DeepDiver1975 possible regression

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Related Issue
fixes https://github.com/owncloud/core/issues/27344

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by PVince81** [comment]
Shouldn't signed apps appear approved instead of no label at all ? Note that hiding experimental might not be enough, there is also a sort based on this state.

The actual bug is that apps that used to be recognized as signed in 9.1 aren't any more in 10.0, so I suspect it's rather a bug in the verification code.

**Comment by DeepDiver1975** [comment]
> Shouldn't signed apps appear approved instead of no label at all ? Note that hiding experimental might not be enough, there is also a sort based on this state.

I'll have a look into this ....

**Comment by PVince81** [comment]
Discussion here, apparently this is not possible to bring it back in the old form as it was dependent on the app store. But maybe we can use a new approach with the integrity signature which is based on the developer's key: https://github.com/owncloud/core/issues/27344#issuecomment-288755206

File: lib/private/legacy/app.php [hunk 1]
```diff
@@ -864,6 +865,13 @@ public static function listAllApps() {
 					$info['level'] = self::officialApp;
 					$info['removable'] = false;
 				} else {
+					$result = \OC::$server->getIntegrityCodeChecker()->verifyAppSignature($app, '', true);
```
Comment (PVince81) [inline lib/private/legacy/app.php]: Great. Do we cache this info somewhere ? Because integrity check can be slow sometimes...

Comment (PVince81) [inline lib/private/legacy/app.php]: I guess people with redis or APC will have their app list + info cached anyway so could be fine

Comment (DeepDiver1975) [inline lib/private/legacy/app.php]: the checker has a cache within

**Comment by PVince81** [comment]
Tested with signed customgroups app, works 👍

**Comment by PVince81** [comment]
Rebased for CI.

**Pull request merged**